### PR TITLE
expose max d0 setting

### PR DESCRIPTION
**Pull request by madbaron**
Hi all,

This PR exposes the parameter regulating the maximum track d0 to allow dedicated long lived particle studies.
(And a bunch of whitespace changes made by VS code...)

Cheers,
Federico

### CODE REVIEW

File: src/ACTSSeededCKFTrackingProc.cxx [hunk 1]
```diff
@@ -359,37 +376,36 @@ void ACTSSeededCKFTrackingProc::processEvent( LCEvent* evt )
...
-  finderCfg.cotThetaMax = 7.40627;  // 2.7 eta;
+  finderCfg.cotThetaMax = 7.40627; // 2.7 eta;
   finderCfg.sigmaScattering = _seedFinding_sigmaScattering;
   finderCfg.radLengthPerSeed = _seedFinding_radLengthPerSeed;
   finderCfg.minPt = _seedFinding_minPt * Acts::UnitConstants::MeV;
   finderCfg.bFieldInZ = (*magneticField()->getField(zeropos, magCache))[2];
-  finderCfg.beamPos = {0,0};
-  finderCfg.impactMax = 3;
+  finderCfg.beamPos = {0, 0};
+  finderCfg.impactMax = _CKF_d0CutOff;
```
Comment (kkrizka) [inline src/ACTSSeededCKFTrackingProc.cxx:408]: `* Acts::UnitConstants::mm` to make the units explicit.

File: ACTSTracking/ACTSSeededCKFTrackingProc.hxx [hunk 2]
```diff
@@ -69,23 +67,22 @@ class ACTSSeededCKFTrackingProc : public ACTSProcBase
...
+  double _initialTrackError_phi;
+  double _initialTrackError_relP;
+  double _initialTrackError_lambda;
+  double _initialTrackError_time = 100 * Acts::UnitConstants::ns; // No Marlin default
 
-  double  _CKF_chi2CutOff            =15;
-  int32_t _CKF_numMeasurementsCutOff =10;
+  double _CKF_chi2CutOff = 15;
+  int32_t _CKF_numMeasurementsCutOff = 10;
+  double _CKF_d0CutOff = 3;
```
Comment (kkrizka) [inline ACTSTracking/ACTSSeededCKFTrackingProc.hxx:78]: `* Acts::UnitConstants::mm` to set explicit units.

**Review by kkrizka — CHANGES_REQUESTED** [review]

**Comment by madbaron** [comment]
Hi @kkrizka,

I've implemented your suggestions (and now grouped the d0 cut together with the other seeding parameters).
In the original PR I didn't undo the white space changes, because I checked them and they all look for the better, but please insist if you disagree and I'll get rid of them!

File: src/ACTSSeededCKFTrackingProc.cxx [hunk 3]
```diff
@@ -131,84 +131,92 @@ ACTSSeededCKFTrackingProc::ACTSSeededCKFTrackingProc() : ACTSProcBase("ACTSSeede
                              _seedFinding_minPt,
                              _seedFinding_minPt);
 
+  registerProcessorParameter("SeedFinding_impactMax",
```
Comment (kkrizka) [inline src/ACTSSeededCKFTrackingProc.cxx:134]: Could you please upper case the first letter `"SeedFinding_ImpactMax"` for consistency with the other settings?

**Comment by kkrizka** [comment]
Hi @madbaron ,

Code formatting is very subjective. Unless there is an official style being enforced by a CI, I would prefer that formatting changes are kept out of unrelated (to formatting) PR's.

Removing extra space at the end of a line and consistency changes are welcome though.

--
Karol Krizka

**Comment by madbaron** [comment]
Finally coming back to this :D
I have capitalised the "I" as requested, as well as added an additional handle to steer the extrapolation direction.
@kkrizka, please have a look and let me know if you'd like me to make some other style change.

The only thing I didn't undo is the formatting cleanup, because I don't know how to do it :D

Cheers,
Federico

**Review by kkrizka — APPROVED** [review]

**Pull request merged**